src/utils/component/deleteByKey.ts

```diff
@@ -0,0 +1,32 @@
+import { gql, gqlRequest } from "../../graphql";
+
+export const deleteComponentByKey = async (key: string) => {
+  // Retrieve the existing signature of the component if it exists.
+  const result = await gqlRequest({
+    document: gql`
+      query component($key: String!) {
+        components(key: $key) {
+          nodes {
+            id
```

Comment:
Comment seems inaccurate. I think this probably should filter a bit more defensively too to at least look only at private components.